yolo_to_yolo/godot_reader.py

```diff
@@ -0,0 +1,88 @@
+from pathlib import Path
+from typing import Iterable, Generator
+
+import numpy as np
+from PIL import Image
+
+from yolo_to_yolo.generic_reader import GenericYoloReader
+
+from .data_types import YoloImageData, YoloLabel, YoloBbox, DatasetDescriptor
+from .yolo_io_types import Task, PredictionTask
+
+
+class GodotMultiLabelReader(GenericYoloReader):
+    """
+    Reader for YOLO training data.
+
+    Example:
+        reader = YoloReader("YOLO_DATASET/data.yaml")
+        for yolo_image_data, task in reader:
+            ...
+    """
+    def __init__(
+        self,
+        yaml_path: Path,
+    ) -> None:
+        self.prediction_task = PredictionTask.CLASSIFICATION
+
+        self.yaml_path = yaml_path
+
+        self.descriptor = DatasetDescriptor.from_yaml(self.yaml_path)
+        # same as superclass constructor but without path checking because we need to swap order
+        # of task and directory (e.g. train/images -> images/train)
+
+    def read(
+        self,
+        tasks: tuple[Task, ...] = (Task.TRAIN, Task.VAL, Task.TEST),
+        img_file_pattern: str = "*.png"
+    ) -> Generator[YoloImageData, None, None]:
+        id = 0
+        for task in tasks:
```

Comment:
This is trivially parallelizable, in the exact same way that the original `YoloReader` did it. As a matter of fact, extracting the path enumeration to a helper function would allow you to put the `read` function into the base class, then you only need to have a function for the worker task and a function to get the paths. 

If you'd rather move on quickly, I can take care of that.

Reply:
Not gonna do this, IMO not worth the effort.